gpt4all-bindings/golang/binding.cpp

```diff
@@ -19,46 +16,27 @@
 #include <iostream>
 #include <unistd.h>
 
-void* load_mpt_model(const char *fname, int n_threads) {
+void* load_model(const char *fname, int n_threads) {
     // load the model
-    auto gptj = llmodel_mpt_create();
-
-    llmodel_setThreadCount(gptj,  n_threads);
-    if (!llmodel_loadModel(gptj, fname)) {
+    llmodel_error new_error{};
+    auto model = llmodel_model_create2(fname, "auto", &new_error);
+    if (model == nullptr ){
+        fprintf(stderr, "%s: error '%s'\n",
```

Comment:
Why don't you forward the error to the Go world?

Reply:
load_model returns a pointer up to now, don't want to stretch this PR more, and keep it scoped

Reply:
Alright